pkg/cluster/k8sres.go

```diff
@@ -514,6 +514,7 @@ func generatePodTemplate(
 	additionalSecretMount string,
 	additionalSecretMountPath string,
 	volumes []v1.Volume,
+	additionalVolumes []acidv1.AdditionalVolume,
```

Comment:
Passing volume[] twice is giving me some headache. The first on was introduced by @zimbatm for the TLS secrets. Maybe that could be mapped to an additionalVolumes too, or it's just renamed tlsVolumes. Anyway, can be done in an extra PR.

Reply:
@FxKu -> would this mean that it's either TLS or additionalVolumes?
Our scenario would require both.

Reply:
no I thought both can be mapped into one array.

Reply:
@FxKu if there is a new functionality available fully covering previously specially handled use case there should be some sort of deprecation. This here seems to be one of those cases where the "old" functionality is simply extended upon and folks simply can move to the new, more feature rich way. 

But this could also be the case for the PR implementing a broader sidecar support (https://github.com/zalando/postgres-operator/pull/890). Why having multiple things doing the same thing and creating ever more logic to "merge" them and also having to maintain these increasingly complicated code paths.

Reply:
The two have been merged - see below. 
It can be pushed directly: https://github.com/zalando/postgres-operator/pull/918
@FxKu and Christian - have a look.
If you want, we can also merge go via @seuf's branch (https://github.com/seuf/postgres-operator/pull/1 instead). It's same code in both cases.